firedrake/output.py

```diff
@@ -651,3 +756,366 @@ def write(self, *functions, **kwargs):
                          'file="%s" />\n' % (time, vtu)).encode('ascii'))
                 # And add footer again, so that the file is valid
                 f.write(self._footer)
+
+
+def _get_format(viewer):
+    h5pyfile = hdf5interface.get_h5py_file(viewer)
+    if all(d in h5pyfile for d in ['/geometry/vertices',
+                                   '/topology/cells',
+                                   '/topology/cones',
+                                   '/topology/order',
+                                   '/topology/orientation']):
+        return ViewerHDF5.Format.HDF5_PETSC
+    elif all(d in h5pyfile for d in ['/labels/celltype',
+                                     '/geometry/vertices',
+                                     '/viz/topology/cells']):
+        return ViewerHDF5.Format.HDF5_XDMF
+    else:
+        raise RuntimeError("Unknown HDF5 file format used in %s", viewer.getFileName())
+    del h5pyfile
+
+
+def _dmload_plex(viewer, name):
+    plex = PETSc.DMPlex()
+    plex.create(comm=viewer.comm)
+    plex.setName(name)
+    format = _get_format(viewer)
+    viewer.pushFormat(format=format)
+    if format == ViewerHDF5.Format.HDF5_PETSC:
+        sfXB = plex.topologyLoad(viewer)
+        plex.coordinatesLoad(viewer)
+        plex.labelsLoad(viewer)
+        plex.setOptionsPrefix("loaded_")
+        plex.viewFromOptions("-dm_view")
+    elif format == ViewerHDF5.Format.HDF5_XDMF:
+        plex.load(viewer=viewer)  # TODO: load labels, too.
+        plex.setOptionsPrefix("loaded_")
+        plex.viewFromOptions("-dm_view")
+        plex.interpolate()
+        plex.setOptionsPrefix("interpolated_")
+        plex.viewFromOptions("-dm_view")
+        sfXB = None
```

Comment:
Is this a loading format we want to support? After all, we can't really do anything with it.

Reply:
That is right. Removed all related codes and tests.